File: int/ppl/exercises/ruby/wordcount.rb
#!/usr/bin/env ruby

wordcount = {}
File.open 'words' do |file|
  file.each do |line|
    words = line.split
    words.each do |word|
      same = words.select {|x| x == word}
      wordcount[word] = same.length
    end
  end
end

wordcount.each {|k, p| puts "#{p}: #{k}" if p > 1}

File: int/cc/coursework/wordcount.py
#!/usr/bin/env python

from mrjob.job import MRJob

class MRWordFrequencyCount(MRJob):

    def mapper(self, _, line):
        yield "chars", len(line)
        yield "words", len(line.split())
        yield "lines", 1

    def reducer(self, key, values):
        yield key, sum(values)

if '__main__' == __name__:
    MRWordFrequencyCount.run()

# word word word

